src/components/options-menu/options-menu.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { ChangeEvent, useCallback, useMemo, useRef, useState } from "react";
+import { AnalogStickSize, AppOptions, ColorOption, UrlOptions } from "../../services/url-options.service";
+import './options-menu.scss';
+
+const lettersOnly: RegExp = /^[a-zA-Z]/;
+
+export default function OptionsMenu(props: OptionsMenuProps): React.ReactElement {
+
+  const [formOptions, setFormOptions] = useState<OptionsMenuState>({
+    wheelImgUrl: {
+      value: props.appOptions.wheelImgUrl
+    },
+    wheelColor: {
+      chooser: (Object.values(ColorOption) as string[]).includes(props.appOptions.wheelColor) ? "predefined" : 'custom',
+      value: props.appOptions.wheelColor
+    },
+    analogStickFillColor: {
+      chooser: (Object.values(ColorOption) as string[]).includes(props.appOptions.analogStickFillColor) ? "predefined" : 'custom',
+      value: props.appOptions.analogStickFillColor
+    },
+    analogStickFillOpacity: {
+      value: props.appOptions.analogStickFillOpacity
+    },
+    analogStickBorderColor: {
+      chooser: (Object.values(ColorOption) as string[]).includes(props.appOptions.analogStickBorderColor) ? "predefined" : 'custom',
+      value: props.appOptions.analogStickBorderColor
+    },
+    analogStickSize: {
+      value: props.appOptions.analogStickSize
+    }
+  });
+
+  const customUrlTextAreaRef = useRef<HTMLTextAreaElement>(null);
+  const [customUrlCopyBtnText, setCustomUrlCopyBtnText] = useState<string>('Copy');
+
+  const generateUrlParams = useCallback(() => {
+    const paramMap = {
+      [UrlOptions.wheelImgUrlKey]: formOptions.wheelImgUrl.value,
+      [UrlOptions.wheelColorKey]: formOptions.wheelColor.value,
+      [UrlOptions.analogStickSizeKey]: formOptions.analogStickSize.value,
+      [UrlOptions.analogStickFillColorKey]: formOptions.analogStickFillColor.value,
+      [UrlOptions.analogStickFillOpacityKey]: formOptions.analogStickFillOpacity.value,
+      [UrlOptions.analogStickBorderColorKey]: formOptions.analogStickBorderColor.value
+    };
+    return "?" + Object.entries(paramMap).filter(([key, value]) => value != null && value !== '').map(([key, value]) => key + "=" + value).join("&");
```

Comment:
use !==

Reply:
Also is there a utility that basically does this that we can import?

Reply:
> use !==

Using `!= null ` will account for null or undefined, and `!== ''` will filter out any empty string values. the only values that can exist here are `undefined` or a string, but I am just trying to avoid any strange behavior with '0' or basically any other falsey values.

> Also is there a utility that basically does this that we can import?

There probably is, but not sure off the top of my head. When I tried looking it up, most things suggest doing it just like we have it, so I'm guessing that's where we go it. Once thing I can probably do is use `encodeURIComponent` on the key and value though, which should make it more url safe.

Reply:
okay, makes sense.